### PR TITLE
Issue 4416: Remove deprecated client APIs

### PR DESCRIPTION
**Pull request by tkaitchuck**
**Change log description**  
* Removes `ClientFactory` (Replaced by EventStreamClientFactory)
* Removes `beginTxn()` and `getTxn()` methods from the EventStreamWriter. (Replaced by methods on TransactionalEventStreamWriter)
* Removes the ByteStreamClient interface. (Replaced by ByteStreamClientFactory)

**Purpose of the change**  
Fixes https://github.com/pravega/pravega/issues/4416

**What the code does**  
Removes APIs deprecated in https://github.com/pravega/pravega/pull/3126
No other functional changes.

### CODE REVIEW

**Comment by tkaitchuck** [comment]
@shrids  Build #1949 went through.

File: client/src/main/java/io/pravega/client/stream/ReaderGroup.java [hunk 1]
```diff
@@ -10,9 +10,7 @@
 package io.pravega.client.stream;
 
 import com.google.common.annotations.Beta;
```
Comment (RaulGracia) [inline client/src/main/java/io/pravega/client/stream/ReaderGroup.java]: Can we also remove those API calls tagged with "Beta"? For instance `generateStreamCuts` has been there for the last 2 major versions at least. The same for the Byte API related methods (I also remember some Beta annotation there). I think that we can remove these `Beta` annotations at this point as many of these calls have been tested for quite a while so far. Also, I still see a deprecated call in this clasee: `resetReadersToCheckpoint`. Can we get rid of this one also in this PR to help users to use the right methods from 0.7 onwards?

**Comment by tkaitchuck** [comment]
> System test run `/test-pravega-continuous-1/442/` was successful with this change set.
> 
> Can we remove the the following deprecated apis too in this PR?
> 
> * `io.pravega.client.batch.BatchClient`
> * `io.pravega.client.batch.StreamInfo`
> * `io.pravega.client.stream.StreamConfiguration.StreamConfigurationBuilder#scope`
> * `io.pravega.client.stream.StreamConfiguration.StreamConfigurationBuilder#streamName`
> * `io.pravega.client.stream.ReaderGroup#resetReadersToCheckpoint`
> 
> Also, we can remove redundant @SuppressWarnings("deprecation") in `client/src/main/java/io/pravega/client/stream/impl/ClientFactoryImpl.java`

Done.

File: client/src/main/java/io/pravega/client/stream/ReaderGroup.java (hunk 1 above)
Comment (tkaitchuck) [reply client/src/main/java/io/pravega/client/stream/ReaderGroup.java]: We should do that in a different PR.

**Comment by RaulGracia** [comment]
Once merged, we need to focus on at least (but not only):
- https://github.com/pravega/flink-connectors/issues/304
- https://github.com/pravega/pravega-samples/pull/231
- https://github.com/pravega/video-samples/issues/10
- https://github.com/pravega/hadoop-connectors/issues/106

**Pull request merged**